### PR TITLE
[settings] Add nova setting structs and one trait

### PR DESCRIPTION
**Pull request by coler706**

### CODE REVIEW

File: src/settings/settings_structs.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
+use super::settings_traits::*;
+use crate::rhi::*;
+pub struct Semver {
+    major: u32,
+    minor: u32,
+    patch: u32,
+}
+
+/*
```
Comment (DethRaid) [inline src/settings/settings_structs.rs]: Please use the Rust commenting style, explained at https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html?highlight=document#making-useful-documentation-comments

File: src/settings/mod.rs [hunk 2]
```diff
@@ -4,3 +4,12 @@
 //! possibly by reading from an on-disk configuration file or asking the end user for settings. The settings are then
 //! used throughout Nova for various purposes. While most of these settings will be pretty technical and only useful to
 //! the application developer, a few of these, such as the API to use, will likely be more interesting for the end user
+
+mod settings_enums;
```
Comment (DethRaid) [inline src/settings/mod.rs]: You don't need to split up files like this, especially if one of the files is empty. Use whatever file structure you think makes the most sense

**Comment by cwfitzgerald** [comment]
@coler706 just a ping on the status of this?

**Comment by DethRaid** [comment]
It's been a month, and near as I can tell @coler706 abandoned thie PR. Closing it so we can move on with our lives

**Comment by coler706** [comment]
Sincere Apologies. I thought I would be free enough this summer to get back to work on this project. But sadly that didn't turn out. :(

**Pull request closed without merging**